ocaml/typing/typemod.ml

```diff
@@ -802,7 +802,7 @@ let merge_constraint initial_env loc sg lid constr =
         let mty = Mtype.scrape_for_type_of ~remove_aliases sig_env mty in
         let mty =
           remove_modality_and_zero_alloc_variables_mty sig_env
-            ~zap_modality:Mode.Modality.Value.zap_to_floor mty
```

Comment:
I can't think of any correct use of `zap_to_floor`. And a quick search seems to find this use (which is wrong) and also a use in printing (which I believe is also wrong). @riaqn -- what do you think? I think we always want `zap_to_id`.

Reply:
Generating .cmi without .mli or printing back the type of a user-written module in REPL, describes the structure. In these cases, the inferred modalities default to the strongest possible.

Reply:
Yeah that's right. For monadic axes, the modalities are of the form `join_with_c`. And the identity modality is `join_with_min`.

Reply:
@goldfirere can I resolve this?